File: Hello_name/hello_name.go
package main

import (
	"fmt"
)

func main() {
	var nome string

	// Solicite ao usuário que insira seu nome
	fmt.Print("Digite seu nome: ")

	// processamento
	fmt.Scanln(&nome)

	// saida de dados
	fmt.Printf("Olá, %s!\n", nome)
}

File: Hello_name/hello_name.cpp
#include <iostream>
#include <string>

int main() {
    std::string nome;

    // Solicite ao usuário que insira seu nome
    std::cout << "Digite seu nome: ";
    std::cin >> nome;

    // Imprima a saudação com o nome
    std::cout << "Olá, " << nome << "!" << std::endl;

    return 0;
}
